src/Mod/Draft/draftobjects/patharray.py

```diff
@@ -206,55 +236,100 @@ def calculatePlacementsOnPath(shapeRotation, pathwire, count, xlate, align):
+            shapeRotation, path[iend], offset, pt, xlate, align, normal, mode, forceNormal))
 
         travel += step
 
     return placements
 
-
-
-def calculatePlacement(globalRotation, edge, offset, RefPt, xlate, align, normal=None):
-    """Orient shape to tangent at parm offset along edge."""
+def calculatePlacement(globalRotation, edge, offset, RefPt, xlate, align, normal=None, 
+        mode = 'Original', overrideNormal=False):
+    """Orient shape to a local coord system (tangent, normal, binormal) at parameter offset (normally length)"""
     import functools
-    # http://en.wikipedia.org/wiki/Euler_angles
-    # start with null Placement point so translate goes to right place.
+    # http://en.wikipedia.org/wiki/Euler_angles  (previous version)
+    # http://en.wikipedia.org/wiki/Quaternions
+    # start with null Placement point so _tr goes to right place.
     placement = App.Placement()
     # preserve global orientation
     placement.Rotation = globalRotation
 
     placement.move(RefPt + xlate)
-
     if not align:
         return placement
 
     nullv = App.Vector(0, 0, 0)
-
-    # get a local coord system (tangent, normal, binormal) at parameter offset (normally length)
-    t = edge.tangentAt(getParameterFromV0(edge, offset))
-    t.normalize()
+    defNormal = App.Vector(0.0, 0.0, 1.0)
+    if not normal is None:
+        defNormal = normal
 
     try:
-        n = edge.normalAt(getParameterFromV0(edge, offset))
-        n.normalize()
-        b = (t.cross(n))
-        b.normalize()
-    # no normal defined here
-    except App.Base.FreeCADError:
-        n = nullv
-        b = nullv
-        App.Console.PrintMessage(
-            "Draft PathArray.orientShape - Cannot calculate Path normal.\n")
-
-    priority = "ZXY"        #the default. Doesn't seem to affect results.
-    newRot = App.Rotation(t, n, b, priority)
+        t = edge.tangentAt(getParameterFromV0(edge, offset))
+        t.normalize()
+    except:
+        _msg("Draft CalculatePlacement - Cannot calculate Path tangent. Copy not aligned\n")
+        return placement
+
+#Original mode is the historic "Align" for old (v0.18) documents.  It is not 
+#really the Fernat alignment. Uses the normal parameter from getNormal (or the
+#default) as a constant - it does not calculate curve normal.
+#X is curve tangent, Y is normal parameter, Z is (X x Y)
+
+#Tangent mode is similar to Original, but includes a pre-rotation (in execute) to 
+#align the Base object's X to the TangentVector, then X follows curve tangent, 
+#normal input parameter is the Z component.
+
+#If the ForceVertical option is applied, the normal parameter from getNormal is 
+#ignored, and X is curve tangent, Z is VerticalVector, Y is (X x Z)
+
+#Frenet mode orients the copies to a coordinate system along the path.
+# X is tangent to curve, Y is curve normal, Z is curve binormal. 
+# if normal can not be computed (ex a straight line), the default is used.
```

Comment:
This should be indented just like the code. A big explanation like this should be in the docstring of the function.

Reply:
Done.